src/site.rkt

```diff
@@ -1517,11 +1515,15 @@
          [st (hash-set
               st
               (string->symbol
-               (format "ring:~a" (hash-ref ht 'ring (lambda () 2)))) #t)]
+               (format "ring:~a" (hash-ref ht 'ring 2)))
+              #t)]
          [st (for/fold ([st st])
-                       ([a (in-list (string-split (hash-ref ht 'author (lambda () ""))))])
+                       ([a (in-list (string-split (hash-ref ht 'author "")))])
                (hash-set
-                st (string->symbol (format "author:~a" a)) #t))]
+                st
+                (string->symbol
+                 (display-name-tag (author-display-name a))
+                 #t)))]
```

Comment:
While working on this, I discovered that currently, though email addresses aren't displayed, they are still searchable as tags: see e.g. <https://pkgd.racket-lang.org/pkgn/search?q=&tags=author%3Aphilip%40philipmcgrath.com+>.

In this patch I tried to use the obfuscated display names, instead, but the comment above `compute-search-terms` suggests that there might be related behavior in `pkg-index` somewhere. Possibly also the search implementation could translate author queries to also match the obfuscated form. I haven't tracked down where all of these pieces go yet, which is the main reason I've marked this as a draft.